src/ui/verificationGutterStatusView.ts

```diff
@@ -0,0 +1,345 @@
+  private areParamsOutdated(params: IVerificationGutterStatusParams): boolean {
+    const documentPath = getVsDocumentPath(params);
+    const previousVersion = this.dataByDocument.get(documentPath)?.version;
+    return (previousVersion !== undefined && params.version !== undefined
+      && params.version < previousVersion);
+  }
+
+  // Entry point when receiving IVErificationStatusGutter
+  private updateVerificationStatusGutter(params: IVerificationGutterStatusParams): void {
+    if(this.areParamsOutdated(params)) {
+      return;
+    }
+    const documentPath = getVsDocumentPath(params);
+    const ranges = this.getRangesOfLineStatus(params);
+
+    const newData: LinearVerificationGutterStatus = {
+      decorations: ranges,
+      version: params.version };
+
+    this.setDisplayedVerificationStatusGutter(documentPath, newData);
+  }
+
+  // Gets the ranges associated to a line verification status
+  // Adds an empty array if necessary
+  private getRanges(ranges: Map<LineVerificationStatus, Range[]>, status: LineVerificationStatus): Range[] {
+    let r = ranges.get(status);
+    if(r === undefined) {
+      r = [];
+      ranges.set(status, r);
+    }
+    return r;
+  }
+
+  // Given current data and previous data, should we delay the display of the new data?
+  private mustBeDelayed(ranges: Map<LineVerificationStatus, Range[]>, previousRanges: Map<LineVerificationStatus, Range[]>): boolean {
+    return (this.getRanges(ranges, LineVerificationStatus.ResolutionError).length >= 1
+            && this.getRanges(previousRanges, LineVerificationStatus.ResolutionError).length === 0)
+        || (obsoleteLineVerificationStatus.some(status => this.getRanges(ranges, status).length >= 1)
+            && verifyingLineVerificationStatus.every(status => this.getRanges(ranges, status).length === 0)
+            && obsoleteLineVerificationStatus.every(status => this.getRanges(previousRanges, status).length === 0));
+  }
+
+  // Assigns the line verification gutter statuses to the given document.
+  // Launches the animation if necessary, including delaying if necessary
+  private setDisplayedVerificationStatusGutter(documentPath: string, newData: LinearVerificationGutterStatus) {
+    const previousValue = this.dataByDocument.get(documentPath);
+    const ranges = newData.decorations;
+    if(this.animationCallback !== undefined) {
+      clearInterval(this.animationCallback);
+    }
+    const previousRanges = previousValue === undefined ? VerificationGutterStatusView.emptyLinearVerificationDiagnostics : previousValue.decorations;
+    if(this.mustBeDelayed(ranges, previousRanges)) {
+      // Delay resolution errors so that we don't interrupt the verification workflow if they are corrected fast enough.
+      this.animationCallback = setTimeout(() => {
+        this.dataByDocument.set(documentPath, newData);
+        this.refreshDisplayedVerificationGutterStatuses(window.activeTextEditor);
+      }, DELAY_IF_RESOLUTION_ERROR);
+    } else {
```

Comment:
Thanks for the tip, that's interesting ! I did not know about the debounce concept.
This case seems more complicated than debouncing because there are animations too, and timeout is not constant.
I did refactor the code though so that it's more legible. Also, I don't think async/await would increase legibility there, to me the counter-example implementation seems cryptic compared to a simple setTimeout/setInterval I implemented there.